src/Http/Routing/src/Patterns/RoutePatternFactory.cs

```diff
@@ -1084,6 +1084,108 @@ public static RoutePatternParameterPolicyReference ParameterPolicy(string parame
         return ParameterPolicyCore(parameterPolicy);
     }
 
+    internal static RoutePattern Combine(RoutePattern left, RoutePattern right)
```

Comment:
This looks allocatey

Reply:
I'm surprised you say that considering the optimizations. The `RawText` string and the `RoutePattern` itself are the only allocations in most cases.